tests: drivers: counter: fix skip test check

manuargue opened the pull request:
When running this test with multiple counter instances that support different capabilities, if one of the instances does not support the test required capabilities, the test will be immediately skipped not giving the chance to run to the subsequent instances.
Fix this by marking the test as skipped, only of all counter instances under test were skipped.

Fixes #74358

All instances skipped:
```
Running TESTSUITE counter_basic
===================================================================
START - test_all_channels
Skipped for pit0_channel@0
Skipped for pit0_channel@1
 SKIP - test_all_channels in 0.207 seconds
===================================================================
```

Some instances skipped, others executed:
```
===================================================================
START - test_all_channels
Skipped for pit0_channel@0
Testing stm@76200000
Testing stm@76210000
Testing stm@76020000
Testing stm@76030000
 PASS - test_all_channels in 0.636 seconds
===================================================================
```

RichardSWheatley commented on tests/drivers/counter/counter_basic_api/src/test_counter.c:
```diff
@@ -190,12 +192,15 @@ static void test_all_instances(counter_test_func_t func,
 			TC_PRINT("Skipped for %s\n", devices[i]->name);
-			ztest_test_skip();
+			devices_skipped++;
 		}
```
Much better, thank you!

manuargue committed:
tests: drivers: counter: fix skip tests check

When running this test with multiple counter instances enabled that
support different capabilities, if one of the instances does not
support the test required capabilities, the test will be
immediately skipped not giving the chance to run to the subsequent
instances.

Fix this by marking the test as skipped, only of all counter instances
under test were skipped.

Fixes #74358

Signed-off-by: Manuel Argüelles <marguelles.dev@gmail.com>
